firmware/application/radio_state.hpp

```diff
@@ -28,38 +28,47 @@
 #include "receiver_model.hpp"
 #include "transmitter_model.hpp"
 
-/* Stashes current radio bandwidth and sampling rate.
- * Inits to defaults, and restores previous when destroyed.
- * The idea here is that an app will eventually call enable
- * on the model which will sync all of the model state into
- * the radio. We tried lazily setting these using the
- * set_configuration_without_update function, but there are
- * still various UI components (mainly Waterfall) that need
- * the radio set in order to load properly.
+/* Reinitialized model with defaults so each app can start
+ * with a clean, default model instance. We tried lazily
+ * setting these using the set_configuration_without_update
+ * function, but there are still various UI components
+ * (e.g. Waterfall) that need the radio set in order to load.
  * NB: This member must be added before SettingsManager. */
 template <typename TModel, TModel* model>
 class RadioState {
    public:
-    RadioState()
-        : RadioState(max2837::filter::bandwidth_minimum, 3072000) {
+    RadioState() {
+        model->initialize();
```

Comment:
Previously the modulation would be maintained when someone switched to the Audio or Microphone app (e.g. from the Looking Glass, Scanner, Recon, or Search app). Would it be possible to maintain this functionality for the Audio & Microphone apps?

Reply:
I think the best way would be to have the caller explicitly set the radio settings as it launches the new app. That way we don’t rely on global state for passing. Need to give that some thinking though because as soon as the next app is instantiated it would have already modified the state. Hmm. Maybe a different constructor that takes some settings would be the best.

Reply:
I think the same approach could address both issues.
Add an additional constructor that takes a model so callers can pass in a the radio state they want to use. This would override the RadioState and app settings.

One thing, I don't recall the Mic app ever respecting bandwidth/modulation/sampling. It only reused the frequency.
I'm not sure what would make sense because there's only Recon that even launches it.

Reply:
Scanner & Recon both launch the Mic app (it would be good if the other radio parameters could be carried over).

Reply:
Ok, Modulation and Step can be overridden when passing to Audio
Modulation can be kind of overridden when passing to MicTX -- MicTX needs some work to reuse more of the "common" stuff from Freqman/ReceiverModel, but that's for a future PR.